### PR TITLE
Fix for issue-1507 illegal Imports.env with a type conflict to the indexer in in…

### PR DESCRIPTION
**Pull request by Canyala**
…dex.d.ts of the loader.

Issue https://github.com/AssemblyScript/assemblyscript/issues/1507

- [x] I've read the contributing guidelines

### CODE REVIEW

File: lib/loader/index.d.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export interface ResultObject {
 
 /** WebAssembly imports with an optional env object and two levels of nesting. */
 export type Imports = {
-  [key: string]: Record<string,unknown>;
+  [key: string]: Record<string,unknown> | undefined;  /** env?: {} not allowed without the '| undefined' anymore */
```
Comment (MaxGraey) [inline lib/loader/index.d.ts:11]:
```suggestion
  [key: string]: Record<string,unknown> | undefined;
```
Could you remove this comment? I think it's unnecessary

Comment (Canyala) [reply lib/loader/index.d.ts]: Done.

**Comment by MaxGraey** [comment]
you should also add yourself to NOTICE file. See contributing guidelines

File: NOTICE [hunk 2]
```diff
@@ -28,6 +28,7 @@ under the licensing terms detailed in LICENSE:
 * Maël Nison <nison.mael@gmail.com>
 * Valeria Viana Gusmao <valeria.viana.gusmao@gmail.com>
 * Gabor Greif <ggreif@gmail.com>
+* Martin Fredriksson <martin.fredriksson@canyala.se>
```
Comment (MaxGraey) [inline NOTICE:31]:
```suggestion
* Martin Fredriksson <martin.fredriksson@vikinganalytics.se>
```

**Comment by dcodeIO** [comment]
Thanks!

**Pull request merged**